### PR TITLE
[WIP] Update example app with latest changes.

### PR DESCRIPTION
**Pull request by alloy**

### CODE REVIEW

File: example/ts/__relay_artifacts__/RemoveTodoMutation.graphql.ts [hunk 1]
```diff
@@ -1,10 +1,7 @@
 /* tslint:disable */
 
 import { ConcreteRequest } from "relay-runtime";
-export type RemoveTodoInput = {
-    readonly id: string;
-    readonly clientMutationId?: string | null;
-};
+export type RemoveTodoInput = {};
```
Comment (sibelius) [inline example/ts/__relay_artifacts__/RemoveTodoMutation.graphql.ts]: Is this the right type?

File: example/ts/__relay_artifacts__/AddTodoMutation.graphql.ts [hunk 2]
```diff
@@ -1,10 +1,7 @@
 /* tslint:disable */
 
 import { ConcreteRequest } from "relay-runtime";
-export type AddTodoInput = {
-    readonly text: string;
-    readonly clientMutationId?: string | null;
-};
+export type AddTodoInput = {};
```
Comment (alloy) [inline example/ts/__relay_artifacts__/AddTodoMutation.graphql.ts]: These changes shouldn’t be happening. Not sure yet why it does.

/cc @koddsson

File: example/ts/__relay_artifacts__/RemoveTodoMutation.graphql.ts (hunk 1 above)
Comment (alloy) [reply example/ts/__relay_artifacts__/RemoveTodoMutation.graphql.ts]: Nope

File: example/ts/__relay_artifacts__/AddTodoMutation.graphql.ts (hunk 2 above)
Comment (koddsson) [reply example/ts/__relay_artifacts__/AddTodoMutation.graphql.ts]: Hmm, this is weird. My app is not doing this but I am using relay 2.0.

I'm off on holiday from work but I could see if I can poke around in this PR if I have some time.

Comment (alloy) [reply example/ts/__relay_artifacts__/AddTodoMutation.graphql.ts]: I’ll give Relay 2 a try tomorrow 👍

File: example/package.json [hunk 3]
```diff
@@ -2,41 +2,38 @@
...
-    "graphql-relay": "^0.5.1",
+    "express-graphql": "^0.7.1",
+    "graphql": "^14.0.2",
+    "graphql-compiler": "^2.0.0-rc.1",
+    "graphql-relay": "^0.5.5",
     "prop-types": "^15.6.2",
     "react": "^16.4.1",
     "react-dom": "^16.4.1",
-    "react-relay": "https://github.com/alloy/relay/releases/download/v1.6.1-plugin.1/react-relay-1.6.1-plugin.1.tgz",
+    "react-relay": "^2.0.0-rc.1",
```
Comment (sibelius) [inline example/package.json]: does we need to upgrade react and react-dom as well?

related to this https://github.com/facebook/relay/issues/2565#issuecomment-447805467

see https://github.com/facebook/relay/releases/tag/v2.0.0-rc.1

File: example/ts/__relay_artifacts__/AddTodoMutation.graphql.ts (hunk 2 above)
Comment (koddsson) [reply example/ts/__relay_artifacts__/AddTodoMutation.graphql.ts]: I tried debugging this a bit and I found out that this seems to be some issue with some relay part. Not sure what, but the types are passed to the `formatGeneratedModule.js` function empty so it doesn't seem to be any thing we introduced but some sort of dependency mismatch or even a bug on some version of relay we have.

It's really hard to debug the example code when it's referenced via a packed module, I think if I'm going to debug this anymore I'll remove that resolution thing and just require the `lib/` folder of the parent directory so I can edit the node modules of the package with debug `console.log`s to help me debug it. That will also make sure I don't have to build and pack every time I want to change anything.

**Comment by sibelius** [comment]
it is working fine on 1.7.0 as well

**Comment by alloy** [comment]
@sibelius Can you elaborate? iirc this PR was using 1.7.0 and it led to these empty input interfaces 
https://github.com/relay-tools/relay-compiler-language-typescript/pull/79#discussion_r243428838

**Comment by sibelius** [comment]
it is working in all our internal projects, maybe we can generated all of them again

**Comment by alloy** [comment]
Hmm, weird. I'll give it a go 👍

**Pull request merged**